internal/provider/resource_auth_method_oidc.go

```diff
@@ -0,0 +1,515 @@
+			},
+			authmethodOidcDisableDiscoveredConfigValidationKey: {
+				Description: "Disables validation logic ensuring that the OIDC provider's information from its discovery endpoint matches the information here. The validation is only performed at create or update time.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+			},
+			authmethodOidcMaxAgeKey: {
+				Description: "The max age to provide to the provider, indicating how much time is allowed to have passed since the last authentication before the user is challenged again.",
+				Type:        schema.TypeInt,
+				Optional:    true,
+			},
+			authmethodOidcSigningAlgorithmsKey: {
+				Description: "Allowed signing algorithms for the provider's issued tokens.",
+				Type:        schema.TypeList,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+				Optional: true,
+			},
+
+			// OIDC specific immutable and computed parameters
+			authmethodOidcClientSecretHmacKey: {
+				Description: "The HMAC of the client secret returned by the Boundary controller, which is used for comparison after initial setting of the value.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+			},
+			authmethodOidcStateKey: {
+				Description: "Can be one of 'inactive', 'active-private', or 'active-public'. Currently automatically set to active-public.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+			},
+			authmethodOidcCallbackUrlKey: {
+				Description: "The URL that should be provided to the IdP for callbacks.",
+				Type:        schema.TypeString,
+				Optional:    true,
+				Computed:    true,
+			},
+			authmethodOidcIsPrimaryAuthMethodForScope: {
+				Description: "When true, makes this auth method the primary auth method for the scope in which it resides. The primary auth method for a scope means the the user will be automatically created when they login using an OIDC account.",
+				Type:        schema.TypeBool,
+				Optional:    true,
+			},
+
+			TypeKey: {
+				Description: "The type of auth method; hardcoded.",
+				Type:        schema.TypeString,
+				Default:     authmethodTypeOidc,
+				Optional:    true,
+			},
+		},
+	}
+}
+
+func setFromOidcAuthMethodResponseMap(d *schema.ResourceData, raw map[string]interface{}) diag.Diagnostics {
+	d.Set(NameKey, raw[NameKey])
+	d.Set(DescriptionKey, raw[DescriptionKey])
+	d.Set(ScopeIdKey, raw[ScopeIdKey])
+	d.Set(TypeKey, raw[TypeKey])
+
+	if attrsVal, ok := raw["attributes"]; ok {
+		attrs := attrsVal.(map[string]interface{})
+
+		// these are always set
+		d.Set(authmethodOidcStateKey, attrs[authmethodOidcStateKey].(string))
+		d.Set(authmethodOidcIssuerKey, attrs[authmethodOidcIssuerKey].(string))
+		d.Set(authmethodOidcClientIdKey, attrs[authmethodOidcClientIdKey].(string))
+		d.Set(authmethodOidcClientSecretHmacKey, attrs[authmethodOidcClientSecretHmacKey].(string))
+
+		if certs, ok := attrs[authmethodOidcIdpCaCertsKey]; ok {
```

Comment:
nit: consider refactoring these if statements to happen inside a for loop that ranges over all the attribute keys.  For example (not a complete list of keys, but just an idea)
```
for _, k := range []string{authmethodOidcIdpCaCertsKey, authmethodOidcAllowedAudiencesKey } {
    if obj, ok := attrs[k]; ok {
        switch obj.(type) {
            case []interface{}:
                d.Set(k, obj.([]interface{}))
            case bool:
                d.Set(k, obj.(bool))
            case string:
                d.Set(k, obj.(string))
            default:
               // raise some error 
        }
    }
}
```

Reply:
using a for loop like this, makes it easier to check that you didn't miss any fields.

Reply:
I'm happy to make this change in a future PR.

Reply:
You could have key clashes that way, you'd probably want to format it as `attributes.%s` and use k there.

Reply:
In fact arguably this is already a problem here.

Reply:
I'm not sure if it's easy to range over values coming from TF. If it is, then naming these with an `attributes.` prefix and ranging is likely the way to go. Even if it isn't, you should probably ensure things are disambiguated by having the keys that are referencing attributes be prefixed as such.

Reply:
I guess since this is the OIDC specific provider maybe it's not a big deal. I'm not really sure.